packages/components/src/SegmentedControl/src/SegmentedControl.tsx

```diff
@@ -0,0 +1,144 @@
+    /**
+     * The size of the controls.
+     * * @default "sm"
+     */
+    size?: ResponsiveProp<SegmentedControlItemSize>;
+}
+
+const SegmentedControl = (props: SegmentedControlProps, ref: ForwardedRef<HTMLDivElement>) => {
+    [props, ref] = useContextProps(props, ref, SegmentedControlContext);
+
+    const { stylingProps, ...ownProps } = useStyledSystem(props);
+    const {
+        className,
+        children,
+        style,
+        slot,
+        defaultSelectedKey,
+        selectedKey,
+        onSelectionChange,
+        ...otherProps
+    } = ownProps;
+
+    const [selected, setSelected] = useState<Key | undefined>(defaultSelectedKey ?? selectedKey);
+
+    const classNames = clsx(
+        GlobalSegmentedControlCssSelector,
+        cssModule(
+            styles,
+            "hop-SegmentedControl"
+        ),
+        stylingProps.className,
+        className
+    );
+
+    const mergedStyles: CSSProperties = {
+        ...style,
+        ...stylingProps.style
+    };
+
+    const onChange = (values: Set<Key>) => {
+        const [firstKey] = values;
+
+        if (!firstKey) {
+            return;
+        }
+
+        setSelected(firstKey);
+        onSelectionChange?.(firstKey);
+    };
+
+    const selectedKeys = useMemo(() => {
+        if (onSelectionChange) {
+            return selectedKey != null ? [selectedKey] : undefined;
+        }
+
+        return selected ? [selected] : undefined;
+    }, [onSelectionChange, selectedKey, selected]);
+
+    useEffect(() => {
+        const container = ref.current;
+
+        // Code to create sliding animation background between buttons when selecting a new value
+        if (container && (selectedKeys || defaultSelectedKey)) {
+            const { childNodes } = container;
+
+            const childNodesArray = Array.from(childNodes) as HTMLElement[];
+
+            const [firstKey] = selectedKeys ?? [];
+            const key = firstKey ?? defaultSelectedKey;
+            const selectedNode = childNodesArray.find(x => x.getAttribute("data-key") === key);
+
+            if (!selectedNode) {
+                return;
+            }
+
+            container.style.setProperty("--hop-SegmentedControl-item-width", `${selectedNode.offsetWidth}px`);
+            container.style.setProperty("--hop-SegmentedControl-item-offset", `${selectedNode.offsetLeft}px`);
+        }
+    }, [ref, selectedKeys, defaultSelectedKey]);
+
+    return (
+        <ToggleButtonGroup
+            ref={ref}
+            className={classNames}
+            style={mergedStyles}
+            slot={slot ?? undefined}
+            selectedKeys={selectedKeys}
+            defaultSelectedKeys={defaultSelectedKey != null ? [defaultSelectedKey] : undefined}
+            orientation="horizontal"
+            onSelectionChange={onChange}
+            {...otherProps}
+        >
+            <Provider
+                values={[
+                    [SegmentedControlItemContext, otherProps]
```

Comment:
Isn't forwarding all other props to the items a bit weird? if i set an aria-label on the ButtonGroup, it's also forwarded on the items?